src/main/java/ch/wisv/areafiftylan/service/EventServiceImpl.java

```diff
@@ -0,0 +1,122 @@
+package ch.wisv.areafiftylan.service;
+
+import ch.wisv.areafiftylan.dto.EventDTO;
+import ch.wisv.areafiftylan.exception.EventException;
+import ch.wisv.areafiftylan.model.Event;
+import ch.wisv.areafiftylan.model.Team;
+import ch.wisv.areafiftylan.service.repository.EventRepository;
+import com.google.common.base.Strings;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+import java.util.Set;
+
+/**
+ * Created by Sille Kamoen on 27-3-16.
+ */
+@Service
+public class EventServiceImpl implements EventService {
+
+    private EventRepository eventRepository;
+
+    @Autowired
+    public EventServiceImpl(EventRepository eventRepository) {
+        this.eventRepository = eventRepository;
+    }
+
+    @Override
+    public void registerTeam(Long eventId, Team team) {
+        Event event = getEventById(eventId);
+        if (event.getTeamSize() != team.getSize()) {
+            throw new EventException("Team is not of the right size!");
+        }
+
+        if (event.getRegisteredTeams().size() > event.getTeamLimit()) {
+            throw new EventException("Event is full!");
+        }
+
+        if (!event.isOpenForRegistration()) {
+            throw new EventException("Event is not open for registration!");
+        }
+
+        // Check if one of the Team members is already a member
+        boolean duplicateUser = event.getRegisteredTeams().stream().flatMap(t -> t.getMembers().stream())
+                .anyMatch(u -> team.getMembers().contains(u));
+
+        if(duplicateUser){
+            throw new EventException("A Member on your Team is already registered");
+        }
+
+        // We're registering the Team here
+        if (!event.addTeam(team)) {
+            throw new EventException("Something went wrong while registrating");
```

Comment:
Why? And if this is true, I still think we should come up with a way to be able to provide more feedback upon failing instead of "Something went wrong."